code-challenges.js

```diff
@@ -3,44 +3,102 @@
 // --------------------1) Create a function that takes a number as an argument and decides if the number is evenly divisble by three or not.
 // Use the test variables provided.
 
-var num1 = 15
-// Expected output: "15 is divisible by three"
 
-var num2 = 0
-// Expected output: "0 is divisible by three"
+// const divisible = (num) => {
+//     if(num % 3 == 0){
+//         return `${num} is divisible by three`
+//     }else{
+//         return `${num} is not divisible by three`
+//     }
+// }
 
-var num3 = -7
-// Expected output: "-7 is not divisble by three"
+// var num1 = 15
+// // Expected output: "15 is divisible by three"
 
+// var num2 = 0
+// // Expected output: "0 is divisible by three"
 
+// var num3 = -7
+// // Expected output: "-7 is not divisble by three"
+
+// console.log(divisible(num3))
 
 
 
 // --------------------2) Create a function that takes in the randomNouns variable and returns an array with all the words capitalized.
 
-var randomNouns = ["streetlamp", "potato", "teeth", "conclusion", "nephew", "temperature", "database"]
-// Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew", "Temperature", "Database"]
+// var randomNouns = ["streetlamp", "potato", "teeth", "conclusion", "nephew", "temperature", "database"]
+// // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew", "Temperature", "Database"]
+
+
+// const capital = (array) => {
 
+//     let newArray = []
 
+//     for (let index = 0; index < array.length; index++) {
+//         newArray.push(array[index].toUpperCase())        
+//     }
 
+//     return newArray
+// }
 
+// console.log(capital(randomNouns))
 
 // --------------------3) Create a function that takes in the mixedDataArray array and returns an array with ONLY NUMBERS sorted from least to greatest.
 
 var mixedDataArray = [true, 8, "hello", 90, -8, null, 0, 46, 59, 107, "hey!"]
 // Expected output: [-8, 0, 8, 46, 59, 90, 107]
 
 
+const numSort = (array) =>{
+    let numArray = []
+    let finalArray = []
+    // let smaller = -50
+    numArray = array.filter(value => typeof value === 'number')
+
+    finalArray = numArray.sort((a, b) => a - b);
+
+    return finalArray
+
+    // *******************Why doesnt the code below work?************************
+    // for (let index = 0; index < numArray.length; index++) {
+    //     if(numArray[index] > smaller){
+    //         finalArray.push(numArray[index])
+    //         smaller = numArray[index]
```

Comment:
Smaller doesn't seem to be declared correctly.  I'd have to dive deeper into this to see the real reason why its not working.